dataprofiler/tests/profilers/test_column_profile_compilers.py

```diff
@@ -70,19 +71,71 @@ def test_add_profilers(self):
-        col_pro_compilers.BaseColumnProfileCompiler, __abstractmethods__=set())
+        col_pro_compilers.BaseCompiler, __abstractmethods__=set())
     def test_no_profilers_error(self):
         with self.assertRaises(NotImplementedError) as e:
-            col_pro_compilers.BaseColumnProfileCompiler(None)
+            col_pro_compilers.BaseCompiler()
         self.assertEqual("Must add profilers.", str(e.exception))
 
+    @mock.patch.multiple(
+        col_pro_compilers.BaseCompiler, __abstractmethods__=set(),
+        _profilers='mock')
+    def test_no_options_error(self):
+        with self.assertRaisesRegex(NotImplementedError,
+                                    "Must set the expected OptionClass."):
+            col_pro_compilers.BaseCompiler()
+
     def test_update_match_are_abstract(self):
         six.assertCountEqual(
             self,
             {'profile'},
-            col_pro_compilers.BaseColumnProfileCompiler.__abstractmethods__
+            col_pro_compilers.BaseCompiler.__abstractmethods__
         )
 
 
+class TestUnstructuredCompiler(unittest.TestCase):
+
+    @mock.patch('dataprofiler.profilers.unstructured_labeler_profile.'
+                'DataLabeler')
+    @mock.patch('dataprofiler.profilers.unstructured_labeler_profile.'
+                'CharPostprocessor')
+    def test_base(self, *mocks):
+        import pandas as pd
+        from collections import defaultdict
+        df_series = pd.Series(['test', 'hi my name is John Doe. 123-432-1234'])
+
+        time_array = [float(i) for i in range(100, 0, -1)]
+        with mock.patch('time.time', side_effect=lambda: time_array.pop()):
+            compiler = col_pro_compilers.UnstructuredCompiler(df_series)
+
+        expected_dict = {
+            'data_label': {
+                'entity_counts': {
+                    'postprocess_char_level': defaultdict(int),
+                    'true_char_level': defaultdict(int),
+                    'word_level': defaultdict(int)},
+                'times': {'data_labeler_predict': 1.0}},
+            'statistics': {
+                'times': {'vocab': 1.0, 'words': 1.0},
+                'vocab': [' ', '-', '.', '1', '2', '3', '4', 'D', 'J', 'a', 'e',
+                          'h', 'i', 'm', 'n', 'o', 's', 't', 'y'],
+                'word_count': {'123': 1, '1234': 1, '432': 1, 'Doe': 1,
+                               'John': 1, 'hi': 1, 'name': 1, 'test': 1},
+                'words': ['test', 'hi', 'name', 'John', 'Doe', '123', '432',
+                          '1234']}}
+
+        output_profile = compiler.profile
+
+        # because vocab uses a set, it will be random order every time, hence
+        # we need to sort to check exact match between profiles
+        if ('statistics' in output_profile
+                and 'vocab' in output_profile['statistics']):
+            output_profile['statistics']['vocab'] = \
+                sorted(output_profile['statistics']['vocab'])
+
+        self.maxDiff = None
```

Comment:
What is this for?

Reply:
I can remove, but basically it sets it so there's no limit on output of the profile dict if there's a failed assertion. Will detail the entire difference rather than clip if too long